app/javascript/src/NewService/components/ServiceSourceForm.jsx

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import React from 'react'
+
+type Props = {
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  handleFormsVisibility: (event: SyntheticEvent<HTMLInputElement>) => void
+}
+
+const ServiceSourceForm = (props: Props) => {
+  const {isServiceDiscoveryUsable, serviceDiscoveryAuthenticateUrl,
+    handleFormsVisibility} = props
+  return (
+    <form className="formtastic" id="new_service_source">
+      <fieldset className="inputs">
+        <ol>
+          <li className="radio">
+            <label htmlFor="source_manual">
+              <input type="radio" name="source" id="source_manual" value="manual" defaultChecked="defaultChecked" onChange={handleFormsVisibility}/> {' Define manually'}
+            </label>
+          </li>
+          <li className="radio">
+            <label htmlFor="source_discover"
+              className={isServiceDiscoveryUsable ? '' : 'disabled'}>
```

Comment:
this conditionals you could extract them to const classNames, so would be easy to keep track of them before returning the jsx.